File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        def sod(n):
            total = 0
            while n > 0:
                total+= n%10
                n//=10
            return total      
        for i,val in enumerate(nums):
            if sod(val) == i:
                return i
        return -1             
                

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int sod(int n){
        int total = 0;
        while (n > 0){
            total += n % 10;
            n /= 10;
        }
        return total;
    }
    public int smallestIndex(int[] nums) {
        for(int i=0; i<nums.length; i++){
            if (sod(nums[i]) == i)
                return i;
        }
        return -1;
    }
}